crates/hyperqueue/src/worker/hwdetect.rs

```diff
@@ -58,12 +62,26 @@ pub fn detect_generic_resources() -> anyhow::Result<Vec<GenericResourceDescripto
     }
 
     if let Ok(mem) = read_linux_memory() {
-        log::debug!("Mem detected: {}", mem);
+        log::info!("Detected {mem}B of memory ({})", human_size(mem));
         generic.push(GenericResourceDescriptor::sum(MEM_RESOURCE_NAME, mem));
     }
     Ok(generic)
 }
 
+/// Tries to detect available Nvidia GPUs from the `CUDA_VISIBLE_DEVICES` environment variable.
+fn detect_gpus_from_env() -> Option<GenericResourceDescriptor> {
+    if let Ok(devices) = std::env::var("CUDA_VISIBLE_DEVICES") {
+        if let Ok(devices) = parse_comma_separated_numbers(&devices) {
+            log::info!(
+                "Detected GPUs {} from `CUDA_VISIBLE_DEVICES`",
+                format_comma_delimited(&devices)
+            );
+            return Some(GenericResourceDescriptor::list(GPU_RESOURCE_NAME, devices));
```

Comment:
It probably needs some kind of sanity check that indices are unique, otherwise it will break the allocator.
(As I am thinking about it, it should be also done for generic list defined resources)